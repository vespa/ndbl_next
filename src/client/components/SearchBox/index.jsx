import React from 'react';
import css from './style.scss';
import cssMain from '../../../scss/main.scss';

const SearchBox = () => {
  return (
    <div className={css['search-box']}>
      <form>
        <input type="search" className={css['search-box__input']} />
        <button
          type="submit"
          className={`
            ${cssMain.icon}
            ${cssMain.icon__search}
            ${css['search-box__button']}
          `}
        >
        search
        </button>
      </form>
    </div>
  );
};

export default SearchBox;
