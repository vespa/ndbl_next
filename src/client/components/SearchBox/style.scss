.search-box  {
  width: 100%;
  position: relative;
  width: 100%;
  margin-top: 8px;
  &__input {
    width: 100%;
    border: 0;
    height: 30px;
  }
  &__button {
    position: absolute;
    width: 20px;
    display: block;
    overflow: hidden;
    top:0;
    right: 0;
    margin: 8px;
  }
}