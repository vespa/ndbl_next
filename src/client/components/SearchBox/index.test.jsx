import React from 'react';
import { shallow } from 'enzyme';
import SearchBox from './index';

describe('[Component: Input]', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<SearchBox />);
  });
  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
