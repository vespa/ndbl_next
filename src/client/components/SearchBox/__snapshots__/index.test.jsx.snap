// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component: Input] should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchBox />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form>
        <input
          className={undefined}
          type="search"
        />
        <button
          className="
            undefined
            undefined
            undefined
          "
          type="submit"
        >
          search
        </button>
      </form>,
      "className": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            className={undefined}
            type="search"
          />,
          <button
            className="
            undefined
            undefined
            undefined
          "
            type="submit"
          >
            search
          </button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": undefined,
            "type": "search",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "search",
            "className": "
            undefined
            undefined
            undefined
          ",
            "type": "submit",
          },
          "ref": null,
          "rendered": "search",
          "type": "button",
        },
      ],
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form>
          <input
            className={undefined}
            type="search"
          />
          <button
            className="
            undefined
            undefined
            undefined
          "
            type="submit"
          >
            search
          </button>
        </form>,
        "className": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              className={undefined}
              type="search"
            />,
            <button
              className="
            undefined
            undefined
            undefined
          "
              type="submit"
            >
              search
            </button>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": undefined,
              "type": "search",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "search",
              "className": "
            undefined
            undefined
            undefined
          ",
              "type": "submit",
            },
            "ref": null,
            "rendered": "search",
            "type": "button",
          },
        ],
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
